AntWars/Board/ControllableBoardObject.cs

```diff
@@ -0,0 +1,181 @@
+
+        /// <summary>
+        /// Lässt die Ameise nach rechts bewegen.
+        /// </summary>
+        /// <returns>true wenn das Bewegen erfolgreich war, false wenn etwas im Weg ist.</returns>
+        public bool moveRight()
+        {
+            Coordinates newCoords = new Coordinates(Coords.X + 1, Coords.Y);
+            return move(newCoords);
+        }
+
+        /// <summary>
+        /// Lässt die Ameise diagonal nach rechts oben bewegen.
+        /// </summary>
+        /// <returns>true wenn das Bewegen erfolgreich war, false wenn etwas im Weg ist.</returns>
+        public bool moveUpperRight()
+        {
+            Coordinates newCoords = new Coordinates(Coords.X + 1, Coords.Y - 1);
+            return move(newCoords);
+        }
+
+        /// <summary>
+        /// Lässt die Ameise diagonal nach rechts unten bewegen.
+        /// </summary>
+        /// <returns>true wenn das Bewegen erfolgreich war, false wenn etwas im Weg ist.</returns>
+        public bool moveLowerRight()
+        {
+            Coordinates newCoords = new Coordinates(Coords.X + 1, Coords.Y + 1);
+            return move(newCoords);
+        }
+
+        /// <summary>
+        /// Lässt die Ameise nach oben bewegen
+        /// </summary>
+        /// <returns>true wenn das Bewegen erfolgreich war, false wenn etwas im Weg ist</returns>
+        public bool moveUp()
+        {
+            Coordinates newCoords = new Coordinates(Coords.X, Coords.Y - 1);
+            return move(newCoords);
+        }
+
+        /// <summary>
+        /// Lässt die Ameise nach unten bewegen
+        /// </summary>
+        /// <returns>true wenn das Bewegen erfolgreich war, false wenn etwas im Weg ist</returns>
+        public bool moveDown()
+        {
+            Coordinates newCoords = new Coordinates(Coords.X, Coords.Y + 1);
+            return move(newCoords);
+        }
+
+        private bool move(Coordinates to)
+        {
+            if (!canMove() && this.isAnt())
+            {
+                (this as Ants.Ant).die();
+            }
+            else if (!TookAction && board.BoardObjects.move(this, to))
+            {
+                TookAction = true;
+                UnitsGone++;
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Prüfen, ob die Ameise sich noch bewegen kann.
+        /// </summary>
+        /// <returns>True wenn ja, False wenn nicht</returns>
+        public bool canMove()
+        {
+            if (UnitsGone >= MoveRange)
+                return false;
+            return true;
+        }
+
+        /// <summary>
+        /// Benachrichtigt andere Ameise in der Sichtweite * 2.
+        /// </summary>
+        /// <param name="coords">Die Koordinaten welche den anderen Ameisem mitgeteilt werden soll</param>
+        /// <returns>true wenn erfolgreich false wenn noch cooldown ist</returns>
+        public virtual bool notifyOtherAnts(HashSet<Coordinates> coords)
```

Comment:
> Wieso hast du virtual und nicht abstract genommen? 

Weil man dann im Scout und Warrior die Methode auch implementieren müsste.
Stimmt aber, dass die in die Ant gehört